### PR TITLE
 Handle packages with invalid portable frameworks in the file names

### PR DESCRIPTION
**Pull request by joelverhagen**
Address https://github.com/NuGet/Engineering/issues/1381.

Packages with file names matching `portable-*-*` break catalog2lucene. These are no longer accepted by gallery but there are some in PROD.

### CODE REVIEW

File: src/NuGet.Indexing/Extraction/CatalogPackageMetadataExtraction.cs [hunk 1]
```diff
@@ -238,18 +239,14 @@ private void AddSupportedFrameworks()
                         .GetSupportedFrameworks()
                         .ToList();
                 }
-                catch (ArgumentException ex) when (ex.Message.ToLowerInvariant().StartsWith("invalid portable"))
+                catch (Exception ex) when (
+                    (ex is ArgumentException && ex.Message.ToLowerInvariant().StartsWith("invalid portable"))
+                    || ex is FrameworkException
+                    || ex is PackagingException)
```
Comment (scottbommarito) [inline src/NuGet.Indexing/Extraction/CatalogPackageMetadataExtraction.cs]: The change here is that we now ignore `PackagingException`s, correct?

Comment (joelverhagen) [reply src/NuGet.Indexing/Extraction/CatalogPackageMetadataExtraction.cs]: Yes

File: src/NuGet.Indexing/Extraction/CatalogNuspecReader.cs [hunk 2]
```diff
@@ -29,8 +29,9 @@ private static XDocument CreateXDocument(JObject catalogItem)
             var package = new XElement("metadata");
             var metadata = new XElement("metadata");
 
-            // ID
+            // identity
             var id = new XElement("id", (string)catalogItem["id"]);
+            var version = new XElement("version", (string)(catalogItem["verbatimVersion"] ?? catalogItem["version"]));
```
Comment (joelverhagen) [inline src/NuGet.Indexing/Extraction/CatalogNuspecReader.cs]: `verbatimVersion` did not exist in the earliest catalog leaves.

File: tests/NuGet.IndexingTests/Extraction/CatalogPackageMetadataExtractorTests.cs [hunk 3]
```diff
@@ -527,11 +584,20 @@ private static object WithPackageEntry(string fullName)
...
+
+            // Add required properties if they are missing.
+            if (json[MetadataConstants.IdPropertyName] == null)
             {
-                DateParseHandling = DateParseHandling.DateTimeOffset
-            });
+                json[MetadataConstants.IdPropertyName] = "NuGet.Versioning";
+            }
+
+            if (json[MetadataConstants.VerbatimVersionPropertyName]== null)
```
Comment (loic-sharma) [inline tests/NuGet.IndexingTests/Extraction/CatalogPackageMetadataExtractorTests.cs]: `]== null` -> `] == null`

Comment (joelverhagen) [reply tests/NuGet.IndexingTests/Extraction/CatalogPackageMetadataExtractorTests.cs]: Fixed.

**Pull request merged**